doc/spec/pantry-spec.md

```diff
@@ -0,0 +1,66 @@
+# Roux Pantry Spec
+
+Ingredients in the Roux ecosystem are distributed in a **pantry**: a filesystem
+subtree containing one or more ingredients.
+
+For example:
+
+```
+┌── my-project
+│   └── pantry
+│       ├── forms
+│       │   └── index.scss
+│       │   └── index.hbs
+│       │   └── ingredient.md
+│       ├── sidebar
+│       │   └── index.scss
+│       │   └── index.hbs
+│       │   └── ingredient.md
+│       │   └── model.js
+```
+
+Pantries are typically named `pantry` but can be named anything you choose.
+A consuming application will define the `pantryRoot` in its `package.json`:
+
+```
+ "roux": {
+   "pantryRoot": "pantry"
+ }
+```
+
+
+## `pantry.js`
+
+A pantry is defined by the presence of a file named `pantry.js` in the root
+level pantry directory (above, `pantry`). A pantry cannot contain another pantry.
```

Comment:
> "a folder with a pantry.js can not exist as a child of a folder with a pantry.js"

^ yes, that. I'll revise; as above I think I was unclear about _name_ vs _folder name_ and you are right that we're not being clear about the difference between a _pantry_ itself and the npm package a pantry may have; internal pantries of course won't have an npm package.

This kind of goes back to my confusion re: scope naming. Is `@rmn/whateverpantry` an internal or external pantry; does it have an npm package associated with it or not? I'd love to find a way to distinguish between those things in the naming. (Or in pantry.js!)